Provide windows batch file for running ktlint with java 16+

## Command run:

`java -jar tools/ktlint -F --log-level info`

Note that if I remove the `-F` argument it works fine.

Results in: `java.lang.ExceptionInInitializerError`
See below for stack trace

## Expected Behavior

I expected the tool to run to completion

## Observed Behavior
```
C:\Users\niteris\dev\androidmonitor-client>java -jar tools/ktlint -F --log-level info
15:11:10.237 [main] INFO com.pinterest.ktlint.cli.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ExceptionInInitializerError
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:692)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:645)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:422)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.run(KtlintCommandLine.kt:323)
        at com.pinterest.ktlint.Main.main(Main.kt:35)
Caused by: java.lang.ExceptionInInitializerError
        at org.jetbrains.kotlin.com.intellij.util.exception.FrequentErrorLogger.report(FrequentErrorLogger.java:47)
        at org.jetbrains.kotlin.com.intellij.util.exception.FrequentErrorLogger.info(FrequentErrorLogger.java:43)
        at org.jetbrains.kotlin.com.intellij.psi.impl.DebugUtil.handleUnspecifiedTrace(DebugUtil.java:628)
        at org.jetbrains.kotlin.com.intellij.psi.impl.DebugUtil.currentInvalidationTrace(DebugUtil.java:620)
        at org.jetbrains.kotlin.com.intellij.psi.impl.DebugUtil.calcInvalidationTrace(DebugUtil.java:614)
        at org.jetbrains.kotlin.com.intellij.psi.impl.DebugUtil.onInvalidated(DebugUtil.java:585)
        at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.TreeElement.onInvalidated(TreeElement.java:226)
        at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.TreeElement.invalidate(TreeElement.java:328)
        at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.TreeElement.rawRemove(TreeElement.java:312)
        at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.TreeElement.rawReplaceWithList(TreeElement.java:319)
        at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafElement.rawReplaceWithText(LeafElement.java:146)
        at com.pinterest.ktlint.ruleset.standard.rules.BlockCommentInitialStarAlignmentRule.beforeVisitChildNodes(BlockCommentInitialStarAlignmentRule.kt:58)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$1.invoke(RuleExecutionContext.kt:93)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$1.invoke(RuleExecutionContext.kt:92)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:92)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:29)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:100)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:29)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:100)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:29)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:100)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:29)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:100)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:29)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:100)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle-dhiVX_g(SuppressHandler.kt:28)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:99)
        at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRule(RuleExecutionContext.kt:61)
        at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine$format$3.invoke(KtLintRuleEngine.kt:136)
        at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine$format$3.invoke(KtLintRuleEngine.kt:135)
        at com.pinterest.ktlint.rule.engine.internal.VisitorProvider$visitor$3.invoke(VisitorProvider.kt:46)
        at com.pinterest.ktlint.rule.engine.internal.VisitorProvider$visitor$3.invoke(VisitorProvider.kt:44)
        at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine.format(KtLintRuleEngine.kt:135)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.format(KtlintCommandLine.kt:479)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.process(KtlintCommandLine.kt:465)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:64)
        at com.pinterest.ktlint.cli.internal.KtlintCommandLine$lintFiles$3.invoke$lambda$0(KtlintCommandLine.kt:416)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private transient java.lang.Object java.lang.Throwable.backtrace accessible: module java.base does not "opens java.lang" to unnamed module @1b0375b3
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
        at org.jetbrains.kotlin.com.intellij.util.ReflectionUtil.findFieldInHierarchy(ReflectionUtil.java:154)
        at org.jetbrains.kotlin.com.intellij.util.ReflectionUtil.getDeclaredField(ReflectionUtil.java:279)
        at org.jetbrains.kotlin.com.intellij.openapi.util.objectTree.ThrowableInterner.<clinit>(ThrowableInterner.java:88)
        ... 55 more
```

## Steps to Reproduce

Sorry, the code repo is private. But I can share it privately if you'd like. Just reach out and I'll arrange.

## Your Environment

#### .editorconfig

```
[*.{kt,kts}]
ktlint_code_style = android
max_line_length = off
ktlint_standard_no-wildcard-imports = disabled
ktlint_code_style=android_studio
```

### KTLint Version
Version: 0.49 (but same problem is present on 0.48.2)

### Version of gradle

```
------------------------------------------------------------
Gradle 7.5
------------------------------------------------------------

Build time:   2022-07-14 12:48:15 UTC
Revision:     c7db7b958189ad2b0c1472b6fe663e6d654a5103

Kotlin:       1.6.21
Groovy:       3.0.10
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.6 (Oracle Corporation 17.0.6+9-LTS-190)
OS:           Windows 10 10.0 amd64
```

### Operating System and version:

Windows 10. Also happens on Mac

--- a/ktlint-cli/build.gradle.kts
+++ b/ktlint-cli/build.gradle.kts
@@ -48,6 +48,8 @@ val shadowJarExecutable by tasks.registering(DefaultTask::class) {
 
     dependsOn(tasks.shadowJar)
 
+    val isReleaseBuild = !version.toString().endsWith("SNAPSHOT")
+
     // Find the "ktlint-cli-<version>-all.jar" file
     val ktlintCliAllJarFile =
         tasks
@@ -60,17 +62,25 @@ val shadowJarExecutable by tasks.registering(DefaultTask::class) {
     logger.lifecycle("ktlint-cli: Base jar to build self-executable file: ${ktlintCliAllJarFile.absolutePath}")
     inputs.files(ktlintCliAllJarFile)
 
+    val windowsBatchFileInputPath = "$projectDir/src/main/scripts/ktlint.bat"
+    inputs.files(windowsBatchFileInputPath)
+
     // Output is the self-executable file
-    val selfExecutableKtlintPath = "$buildDir/run/ktlint"
-    outputs.files(selfExecutableKtlintPath)
-    if (!version.toString().endsWith("SNAPSHOT")) {
-        // And for releases also the signature file
-        outputs.files("$buildDir/run/ktlint.asc")
+    val outputDirectoryPath = "$buildDir/run"
+    val selfExecutableKtlintOutputPath = "$outputDirectoryPath/ktlint"
+    val selfExecutableKtlintSignatureOutputPath = "$outputDirectoryPath/ktlint.asc"
+    val windowsBatchFileOutputPath = "$outputDirectoryPath/ktlint.bat"
+    outputs.files(selfExecutableKtlintOutputPath)
+    if (isReleaseBuild) {
+        // And for releases also the signature file and a batch file for Windows OS
+        outputs.files(selfExecutableKtlintSignatureOutputPath)
+        outputs.files(windowsBatchFileOutputPath)
     }
 
     doLast {
-        logger.lifecycle("Creating the self-executable ktlint-cli")
-        File(selfExecutableKtlintPath).apply {
+        File(selfExecutableKtlintOutputPath).apply {
+            logger.lifecycle("Creating the self-executable file: $selfExecutableKtlintOutputPath")
+
             // writeText effective replaces the entire content if the file already exists. If appendText is used, the file keeps on growing
             // with each build if the clean target is not used.
             writeText(
@@ -95,11 +105,18 @@ val shadowJarExecutable by tasks.registering(DefaultTask::class) {
 
             setExecutable(true, false)
 
-            if (!version.toString().endsWith("SNAPSHOT")) {
+            if (isReleaseBuild) {
+                logger.lifecycle("Creating the signature file: $selfExecutableKtlintSignatureOutputPath")
                 signing.sign(this)
             }
         }
-        logger.lifecycle("Finished creating the self-executable ktlint-cli")
+        if (isReleaseBuild) {
+            logger.lifecycle("Creating the batch file for Windows OS: $windowsBatchFileOutputPath")
+            File(windowsBatchFileOutputPath).apply {
+                writeText(File(windowsBatchFileInputPath).readText())
+            }
+        }
+        logger.lifecycle("Finished creating output files ktlint-cli")
     }
 }
 

--- a/ktlint-cli/src/main/scripts/ktlint.bat
+++ b/ktlint-cli/src/main/scripts/ktlint.bat
@@ -1,0 +1,7 @@
+@echo off
+REM By default this batch file and the "ktlint.jar" are located in same directory. Please adjust path below when this batch
+REM file and the jar file are located in different directories.
+if not defined JAR_PATH set JAR_PATH=.\ktlint.jar
+
+REM The --add-opens argument is needed for java 16+ (see https://github.com/pinterest/ktlint/issues/1986)
+java --add-opens=java.base/java.lang=ALL-UNNAMED -jar "%JAR_PATH%" %*

